src/pages/create.tsx

```diff
@@ -211,54 +220,58 @@ const CreateProgramPage: Page = () => {
             onSrcChange={setProgramLogoURL}
           />
         </div>
+        <form className="space-y-10">
+          <div>
+            <Text b>Identifier</Text>
+            <div className="h-2" />
+            <div className="inline space-x-2">
+              <Text b2>www.mentorcenter.us/ </Text>
+              <Input
+                title="Identifier"
+                name="Program Identifier"
+                placeholder="mentorship-identifier"
+                value={programIdentifier}
+                className="flex-1"
+                onChange={(e) => {
+                  setProgramIdentifier(e.target.value);
+                }}
+              />
+            </div>
+          </div>
 
-        <div>
-          <Text b>Identifier</Text>
-          <div className="h-2" />
-          <div className="inline space-x-2">
-            <Text b2>www.mentorcenter.us/ </Text>
-            <Input
-              title="Identifier"
-              name="Program Identifier"
-              placeholder="mentorship-identifier"
-              value={programIdentifier}
-              className="flex-1"
-              onChange={(e) => {
-                setProgramIdentifier(e.target.value);
+          <div className="flex">
+            <Button
+              type="reset"
+              variant="inverted"
+              onClick={() => {
+                setStage((prev) => prev - 1);
               }}
-            />
+            >
+              Back
+            </Button>
+            <div className="w-4"></div>
+            <Button
+              type="submit"
+              disabled={loading || programIdentifier.length == 0}
+              onClick={(e) => {
+                e.preventDefault();
+                if (!validateProgramIdentifier(programIdentifier))
+                  setError(
+                    "Your program identifier must be alphanumeric and have at least 4 characters."
+                  );
+                else if (
+                  !loading &&
```

Comment:
I think the `loading` check should be moved tot he beginning, i.e. `if (loading) return`